### DISCUSSION TITLE
Cannot find module '*.mjs' imported from ~/.kit/api/kit.js

### DISCUSSION THREAD

**Discussion by tung-dang** [johnlindquist/kit#714] [Errors] [answered]
Hi, 

I have this error in the `~/.kenv/tmp/my-review-pr/error-2022-01-31-22-21-38.txt` file 

```
Cannot find module '~/.kenv/kenvs/tungkit/.scripts/my-review-pr.mjs' imported from ~/.kit/api/kit.js
Error [ERR_MODULE_NOT_FOUND]: Cannot find module '~/.kenv/kenvs/tungkit/.scripts/my-review-pr.mjs' imported from ~/.kit/api/kit.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:391:11)
    at moduleResolve (node:internal/modules/esm/resolve:893:10)
    at Loader.defaultResolve [as _resolve] (node:internal/modules/esm/resolve:1004:11)
    at Loader.resolve (node:internal/modules/esm/loader:89:40)
    at Loader.getModuleJob (node:internal/modules/esm/loader:242:28)
    at Loader.import (node:internal/modules/esm/loader:177:28)
    at importModuleDynamically (node:internal/modules/esm/translators:115:35)
    at exports.importModuleDynamicallyCallback (node:internal/process/esm_loader:30:14)
    at global.attemptImport (file:///~/.kit/api/kit.js:94:30)
```

When I wrote a script in VSCode, I saw a `mjs` file was auto-generated from the `ts` file. 
I deleted all `mjs` files in `.kenv/.scripts` and hope Kit app will generate `mjs` files again but Kit app does not generate `mjs` files anymore.  Now I guess something is broken. I don't see `mjs` file is auto-generated anymore. 

Could someone please help with this? 
Thank you!

**Comment by johnlindquist** [comment]
Any chance you could record and drop a short video like this? It would really help me see what's going on:

(You can drag/drop videos to auto-upload them)

https://user-images.githubusercontent.com/36073/151899406-7340f7ea-e72d-49bf-a606-6c957b77e418.mp4

**Comment by tung-dang** [comment]
Thanks @johnlindquist for reply! 

Here is the recorded video. 
https://user-images.githubusercontent.com/138803/151904983-f21bfb70-58a1-4a10-8e37-783a73b9f3c1.mov

Do you have a document to explain how Kit app generates `.kenv/scripts/*.ts` file to `.kenv/.scripts/*.mjs`? I will create a new script to see if the `mjs` file generating works or not.

**Reply by tung-dang** [reply]
The recorded video is exported as too low resolution. I will fix it. I'm sorry about that

**Reply by johnlindquist** [reply]
Here's how the "build .ts file on change" works:

> Script Kit is spread across a few repos, mainly `kit` (think of it as the "core"/sdk of the project) and `kitapp` (the "app")


1. `kitapp` watches the `kenv/scripts` dir: https://github.com/johnlindquist/kitapp/blob/main/src/watcher.ts#L109
2. When ts file changes, `kitapp` runs a script that builds the TypeScript file: https://github.com/johnlindquist/kitapp/blob/main/src/build.ts#L10
3. `kit`'s TS build setup:
https://github.com/johnlindquist/kit/blob/main/src/cli/build-ts-script.ts#L4
https://github.com/johnlindquist/kit/blob/main/src/api/kit.ts#L103

**Comment by tung-dang** [comment, marked as answer]
**Solution**: I create a new script and then copy all codes from the previous script. It works well now. 
**Lesion learn**: I should not remove `.kenv/.scripts/*.mjs` file because `mjs` file may not be generated again.
